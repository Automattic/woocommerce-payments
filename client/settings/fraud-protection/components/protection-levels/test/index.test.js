/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import ProtectionLevels from '../index';

let mockAdvancedFraudProtectionSettings = null;

jest.mock( 'wcpay/data', () => ( {
	...jest.requireActual( 'wcpay/data' ),
	useAdvancedFraudProtectionSettings: jest.fn( () => [
		mockAdvancedFraudProtectionSettings,
		jest.fn(),
	] ),
} ) );

describe( 'ProtectionLevels', () => {
	beforeEach( () => {
		global.wcpaySettings = {
			isMultiCurrencyEnabled: '1',
		};
	} );

	it( 'renders', () => {
		mockAdvancedFraudProtectionSettings = [];
		const { container: protectionLevels } = render( <ProtectionLevels /> );

		expect( protectionLevels ).toMatchSnapshot();
	} );
	it( 'renders an error message when settings can not be fetched from the server', () => {
		mockAdvancedFraudProtectionSettings = 'error';
		const { container: protectionLevels } = render( <ProtectionLevels /> );

		expect( protectionLevels ).toMatchSnapshot();
		expect( protectionLevels ).toHaveTextContent(
			/There was an error retrieving your fraud protection settings/i
		);
		expect(
			protectionLevels.getElementsByTagName( 'fieldset' )[ 0 ]
		).toBeDisabled();
	} );
} );
