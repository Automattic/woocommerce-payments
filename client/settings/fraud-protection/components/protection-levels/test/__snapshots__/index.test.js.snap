// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectionLevels renders 1`] = `
<div>
  <fieldset>
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <hr
        class="fraud-protection__list-divider"
      />
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Configure
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;

exports[`ProtectionLevels renders an error message when settings can not be fetched from the server 1`] = `
<div>
  <div
    class="wcpay-inline-notice components-notice is-error"
  >
    <div
      class="components-notice__content"
    >
      There was an error retrieving your fraud protection settings. Please refresh the page to try again.
    </div>
  </div>
  <fieldset
    disabled=""
  >
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <hr
        class="fraud-protection__list-divider"
      />
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Configure
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;

exports[`ProtectionLevels renders with the advanced fraud protection settings selected 1`] = `
<div>
  <fieldset>
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8c0-2.21-1.79-4-4-4s-4 1.79-4 4h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2c-.552 0-1 .448-1 1v2h2v-1.14c1.722-.447 3-1.998 3-3.86zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <hr
        class="fraud-protection__list-divider"
      />
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Edit
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;
